apps/src/applab/applab.js

```diff
@@ -1377,18 +1303,13 @@ Applab.resetElementTray = function (allowEditing) {
   if (designModeClear) {
     designModeClear.style.display = allowEditing ? 'inline-block' : 'none';
   }
-
-};
-
-Applab.isValidElementType = function (type) {
-  for (var prop in Applab.ElementType) {
-    if (type === Applab.ElementType[prop]) {
-      return true;
-    }
-  }
-  return false;
 };
 
+// TODO (brent) I think some of these properties are going to end up having
+// different behaviors based on element type. I think the best way of handling
+// this is to have an onPropertyChange per element that gets the first shot to
```

Comment:
This comment sounds like something that inheritance would solve nicely -- a base class with default onPropertyChange behavior that can be overridden for specific elements.